lib/customer.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'takeaway'
+
+class Customer
+
+  attr_reader :takeaway, :basket
+
+  def initialize(takeaway_klass)
+    @takeaway = takeaway_klass.new
+    @basket = Hash.new(0)
+  end
+
+  def view_menu
+    @takeaway.menu
+  end
+
+  def select_dish(dish, quantity)
+    raise 'Please select a dish from the menu' unless dish_exists?(dish)
+    @basket[dish] += quantity
+    "#{quantity}x #{dish}(s) added to your basket"
+  end
+
+  def order_total
+    @basket.map{|k,v| @takeaway.menu[k] * v }.inject(:+)
+  end
+
+  def place_order
+    raise 'Your basket is empty. Please select a dish' if basket_empty?
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)